Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/eventregistration/checkins/qrcode/DefaultQRCodeVerifierTest.kt

```diff
@@ -0,0 +1,111 @@
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+        every { environmentSetup.appConfigVerificationKey } returns PUB_KEY
+        qrCodeVerifier = DefaultQRCodeVerifier(
+            SignatureValidation(environmentSetup),
+            timeStamper
+        )
+    }
+
+    @Test
+    fun verifyEventSuccess() = runBlockingTest {
+        val time = 2687960 * 1_000L
+        every { timeStamper.nowUTC } returns Instant.ofEpochMilli(time)
+        shouldNotThrowAny {
+            val verifyResult = qrCodeVerifier.verify(ENCODED_EVENT)
+            verifyResult.shouldBeInstanceOf<QRCodeVerifyResult.Success>()
+            verifyResult.apply {
+                event.description shouldBe "CWA Launch Party"
+            }
+        }
+    }
+
+    @Test
+    fun verifyEventStartTimeWaning() = runBlockingTest {
+        val time = 2687940 * 1_000L
+        every { timeStamper.nowUTC } returns Instant.ofEpochMilli(time)
+        shouldNotThrowAny {
+            val verifyResult = qrCodeVerifier.verify(ENCODED_EVENT)
+            verifyResult.shouldBeInstanceOf<QRCodeVerifyResult.StartTimeWarning>()
+            verifyResult.apply {
+                event.description shouldBe "CWA Launch Party"
+            }
+        }
+    }
+
+    @Test
+    fun verifyEventEndTimeWarning() = runBlockingTest {
+        every { timeStamper.nowUTC } returns Instant.now()
+        shouldNotThrowAny {
+            val verifyResult = qrCodeVerifier.verify(ENCODED_EVENT)
+            verifyResult.shouldBeInstanceOf<QRCodeVerifyResult.EndTimeWarning>()
+            verifyResult.apply {
+                event.description shouldBe "CWA Launch Party"
+            }
+        }
+    }
+
+    @Test
+    fun verifyEventWithInvalidKey() = runBlockingTest {
+        every { environmentSetup.appConfigVerificationKey } returns INVALID_PUB_KEY
+        shouldThrow<InvalidQRCodeSignatureException> {
+            qrCodeVerifier.verify(ENCODED_EVENT)
+        }
+    }
+
+    @Test
+    fun eventHasMalformedData() = runBlockingTest {
+        shouldThrow<InvalidQRCodeDataException> {
+            qrCodeVerifier.verify(INVALID_ENCODED_EVENT)
+        }
+    }
+
+    companion object {
+
+        private const val INVALID_PUB_KEY = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEc7DEstcUIRcyk35OYDJ95/hTg" +
+            "3UVhsaDXKT0zK7NhHPXoyzipEnOp3GyNXDVpaPi3cAfQmxeuFMZAIX2+6A5Xg=="
+
+        private const val PUB_KEY = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEafIKZOiRPuJWjKOUmKv7OTJWTyii" +
+            "4oCQLcGn3FgYoLQaJIvAM3Pl7anFDPPY/jxfqqrLyGc0f6hWQ9JPR3QjBw=="
+
+        private const val INVALID_ENCODED_EVENT =
+            "BIPEY33SMVWSA2LQON2W2IDEN5WG64RAONUXIIDBNVSXILBAMNXRBCM4UQARRKM6UQASAHRKCC7CTDWGQ" +
+                "4JCO7RVZSWVIMQK4UPA.GBCAEIA7TEORBTUA25QHBOCWT26BCA5PORBS2E4FFWMJ3U" +
+                "U3P6SXOL7SHUBCA7UEZBDDQ2R6VRJH7WBJKVF7GZYJA6YMRN27IPEP7NKGGJSWX3XQ"
+
+        private const val ENCODED_EVENT =
```

Comment:
I think it would help with understanding the test if we would have the actual decoded event data in the test.

Reply:
Max provided some samples :https://github.com/corona-warn-app/cwa-app-tech-spec/pull/30/commits/728492c139f442275720c0067bfee9b9620a20cf 

But those samples require the new proto-buff models , when we have I will add more tests , for now I will keep it as it is